gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/test/java/io/gravitee/rest/api/management/rest/resource/auth/CockpitAuthenticationResourceTest.java

```diff
@@ -0,0 +1,289 @@
+    }
+
+    @After
+    public void teardown() {
+        System.clearProperty("cockpit.enabled");
+        System.clearProperty("cockpit.connector.ws.ssl.keystore.type");
+        System.clearProperty("cockpit.truststore.path");
+        System.clearProperty("cockpit.truststore.password");
+        System.clearProperty("cockpit.truststore.type");
+        System.clearProperty("cockpit.keystore.path");
+        System.clearProperty("cockpit.keystore.password");
+        System.clearProperty("cockpit.keystore.type");
+        System.clearProperty("cockpit.keystore.key.alias");
+    }
+
+    @Test
+    public void shouldNotAuthenticateIfCockpitIsNotEnabled() {
+        System.clearProperty("cockpit.enabled");
+        final Response response = rootTarget(PATH).queryParam("token", "test").request().get();
+        assertEquals(HttpStatusCode.NOT_FOUND_404, response.getStatus());
+    }
+
+    @Test
+    public void shouldNotAuthenticateIfUserNotFound() {
+        UserEntity userEntity = mockUserEntity();
+
+        String jwt = this.generateJWT(userEntity.getId(), ORGANIZATION_ID, ENVIRONMENT_ID, "audience", null, null, null);
+        when(userService.findBySource(ORGANIZATION_ID, "cockpit", userEntity.getId(), true))
+            .thenThrow(new UserNotFoundException(JWT_SUBJECT));
+        final Response response = rootTarget(PATH).queryParam("token", jwt).request().get();
+        assertEquals(HttpStatusCode.FORBIDDEN_403, response.getStatus());
+    }
+
+    @Test
+    public void shouldNotAuthenticateWithInvalidToken() throws JOSEException {
+        UserEntity userEntity = mockUserEntity();
+        when(userService.findBySource(ORGANIZATION_ID, "cockpit", userEntity.getId(), true)).thenReturn(userEntity);
+        when(authoritiesProvider.retrieveAuthorities(userEntity.getId(), ORGANIZATION_ID, ENVIRONMENT_ID)).thenReturn(Set.of());
+        when(installationAccessQueryService.getConsoleUrl(ORGANIZATION_ID)).thenReturn(rootTarget("console").getUri().toString());
+        final Response response = rootTarget(PATH).queryParam("token", "invalid").request().get();
+        assertEquals(HttpStatusCode.INTERNAL_SERVER_ERROR_500, response.getStatus());
+    }
+
+    @Test
+    public void shouldAuthenticateToConsole() {
+        UserEntity userEntity = mockUserEntity();
+
+        String jwt = this.generateJWT(userEntity.getId(), ORGANIZATION_ID, ENVIRONMENT_ID, "audience", null, null, null);
+        when(userService.findBySource(ORGANIZATION_ID, "cockpit", userEntity.getId(), true)).thenReturn(userEntity);
+        when(authoritiesProvider.retrieveAuthorities(userEntity.getId(), ORGANIZATION_ID, ENVIRONMENT_ID)).thenReturn(Set.of());
+        when(installationAccessQueryService.getConsoleUrl(ORGANIZATION_ID)).thenReturn(rootTarget("console").getUri().toString());
+        final Response response = rootTarget(PATH).queryParam("token", jwt).request().get();
+        assertEquals(HttpStatusCode.OK_200, response.getStatus());
```

Comment:
Is it possible to check the url built in the resource? I'm surprise to see 200 status when in the code when send a redirect response that should have a 307 🤔

Reply:
Yes, I was also surprised during the tests, but at that point the redirection was done and the response came from the [final resources](https://github.com/gravitee-io/gravitee-api-management/pull/7118/commits/4a153515a20d506b7849ed9b94661581dc71ad48#diff-79fec14c50e1ad22c9e2f520854cbbb0cf69af0dfad488d890a626524b2c9fbaR271-R287). 
I couldn't find a way to see the url, but we could add a header or something if necessary.

Reply:
Could you add in that method https://github.com/gravitee-io/gravitee-api-management/blob/master/gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/test/java/io/gravitee/rest/api/management/rest/JerseySpringTest.java#L119 the following line

```
config.property(ClientProperties.FOLLOW_REDIRECTS, false);
```

From what I understood it should disable redirection